task/github-open-pr/0.1/github-open-pr.yaml

```diff
@@ -0,0 +1,126 @@
+    description: |
+      The API path prefix, GitHub Enterprise has a prefix e.g. /api/v3
+    default: ""
+    type: string
+
+  - name: REPO_FULL_NAME
+    description: |
+      The GitHub repository full name, e.g.: tektoncd/catalog
+    type: string
+
+  - name: GITHUB_TOKEN_SECRET_NAME
+    description: |
+      The name of the kubernetes secret that contains the GitHub token, default: github
+    type: string
+    default: github
+
+  - name: GITHUB_TOKEN_SECRET_KEY
+    description: |
+      The key within the kubernetes secret that contains the GitHub token, default: token
+    type: string
+    default: token
+
+  - name: AUTH_TYPE
+    description: |
+      The type of authentication to use. You could use the less secure "Basic" for example
+    type: string
+    default: Bearer
+
+  - name: HEAD
+    description: |
+      The name of the branch where your changes are implemented.
+    type: string
+
+  - name: BASE
+    description: |
+      The name of the branch you want the changes pulled into.
+    type: string
+
+  - name: BODY
+    description: |
+      The body description of the pull request.
+    type: string
+
+  - name: TITLE
+    description: |
+      The title of the pull request.
+    type: string
+
+  steps:
+    - name: open-pr
+      env:
+        - name: GITHUBTOKEN
+          valueFrom:
+            secretKeyRef:
+              name: $(params.GITHUB_TOKEN_SECRET_NAME)
+              key: $(params.GITHUB_TOKEN_SECRET_KEY)
+
+      image: registry.access.redhat.com/ubi8/python-38:1-34.1599745032
+      script: |
+        #!/usr/libexec/platform-python
+
+        """This script will open a PR on Github"""
+
+        import json
+        import os
+        import http.client
+
+        open_pr_url = "$(params.API_PATH_PREFIX)" + "/repos/$(params.REPO_FULL_NAME)/pulls"
+
+        data = {
+            "head": "$(params.HEAD)",
+            "base": "$(params.BASE)",
+            "title": "$(params.TITLE)",
+            "body": "$(params.BODY)"
+        }
+        print("Sending this data to GitHub: ")
+        print(data)
+
+        authHeader = "$(params.AUTH_TYPE) " + os.environ["GITHUBTOKEN"]
+
+        # This is for our fake github server
+        if "$(params.GITHUB_HOST_URL)".startswith("http://"):
+            conn = http.client.HTTPConnection("$(params.GITHUB_HOST_URL)"
+                                              .replace("http://", ""))
+        else:
+            conn = http.client.HTTPSConnection("$(params.GITHUB_HOST_URL)")
+
+        conn.request(
+            "POST",
+            open_pr_url,
+            body=json.dumps(data),
+            headers={
+                "User-Agent": "TektonCD, the peaceful cat",
+                "Authorization": authHeader,
+                "Accept": "application/vnd.github.v3+json ",
+            })
+        resp = conn.getresponse()
+        if not str(resp.status).startswith("2"):
+            print("Error: %d" % (resp.status))
+            print(resp.read())
```

Comment:
I wonder if we should store the created pull request url in the results so that it can be later on reused? Otherwise the PR looks good
cc @chmouel

Reply:
That's certainly a possibility! Let me try to get the pull number in there. It's part of the JSON returned by Github.

Reply:
sounds good 👍🏻

Reply:
Added!